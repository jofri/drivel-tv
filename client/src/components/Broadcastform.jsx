
import { useHistory } from "react-router-dom";
import { useState } from 'react';
import '../styles/style.css';

function BroadcastForm () {

  // Import useHistory for redirect functionality
  const history = useHistory();

  // Creates new joined state for all input feilds
  const [ newBroadcast, setNewBroadcast ] = useState({
    title: '',
    description: '',
    tags: '',
    owner: '',
    isReversed: false,
    youtubePlaylists: '',
  });


  // Function that saves the different inputs (by name attribute) to state
  function handleChange(evt) {
    const value = evt.target.value;
    setNewBroadcast({
      ...newBroadcast,
      [evt.target.name]: value
    });
  }

  // Function that saves checkbox input to state
  function handleChangeCheckbox(evt) {
    let value;
    if (evt.target.value === 'false') value = true;
    else value = false;
    setNewBroadcast({
      ...newBroadcast,
      isReversed: value
    });
  };

  // Function to log new broadcast
  async function createBroadcast (broadcast) {
    // Call backend API
    try {
      const response = await fetch('/api/create-broadcast', {
        method: 'POST',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
        },
        body: JSON.stringify(broadcast) // Stringify to JSON before posting
      });
      if (response.ok) { // If response is ok (200 range)
        const broadcastObj = await response.json(); // Parse JSON response

        // Redirect user to new broadcast page
        history.push(`/b/${broadcastObj.broadcastId}`);

      } else {
        // Else if error, send to error page
        history.push(`/404`);
    };
    } catch(err) {
      console.log(err);
    }
  };

  return <>

          <div className="broadcastFormContainer">
          <h3>Create a new broadcast</h3>

            <form data-testid='form' onSubmit={(e) => {

                  e.preventDefault(); /* Prevents reload of page on submit */

                  // Form validation
                  if (newBroadcast.title === '') {alert('Title required!'); return;}
                  if (newBroadcast.owner === '') {alert('Owner field required!'); return;}
                  if (newBroadcast.youtubePlaylists === '') {alert('You have not specefied any youtube playlists!'); return;}

                  createBroadcast(newBroadcast); /* Postsbroadcast to backend through createEvent function */

                  /* Clears inputs */
                  setNewBroadcast({
                    title: '',
                    description: '',
                    tags: '',
                    owner: '',
                    youtubePlaylists: '',
                    isReversed: false,
                  });
              }}>
                  <label>
                    Broadcast name:<br></br>
                    {/* Stores input in state onChange (everytime something is typed) */}
                    <input data-testid='title' name="title" value={newBroadcast.title} onChange={handleChange} type="text"/>
                  </label>
                  <br></br>
                  <br></br>
                  <label>
                    Description:<br></br>
                    {/* Stores input in state onChange (everytime something is typed) */}
                    <input name="description" value={newBroadcast.description} onChange={handleChange} type="text"/>
                  </label>
                  <br></br>
                  <br></br>
                  <label>
                    Tags:<br></br>
                    {/* Stores input in state onChange (everytime something is typed) */}
                    <input name="tags" value={newBroadcast.tags} onChange={handleChange} type="text"/>
                  </label>
                  <br></br>
                  <br></br>
                  <label>
                    Owner:<br></br>
                    {/* Stores input in state onChange (everytime something is typed) */}
                    <input data-testid='owner' name="owner" value={newBroadcast.owner} onChange={handleChange} type="text"/>
                  </label>
                  <br></br>
                  <br></br>
                  <label>
                    Playlists:<br></br>
                    {/* Stores input in state onChange (everytime something is typed) */}
                    <input data-testid='url' name="youtubePlaylists" value={newBroadcast.youtubePlaylists} onChange={handleChange} type="text"/>
                  </label>
                  <br></br>
                  <br></br>
                  <label>
                    Want to reverse the queue order of the playlists?&nbsp;&nbsp;&nbsp;
                    {/* Stores input in state onChange (everytime something is typed) */}
                    <input name="isReversed"  value={newBroadcast.isReversed} onChange={handleChangeCheckbox} type="checkbox"/>
                  </label>
                  <br></br>
                  <br></br>

                  <button type="submit" value="Create">Create Broadcast</button>

            </form>

          </div>



        </>


}

export default BroadcastForm;