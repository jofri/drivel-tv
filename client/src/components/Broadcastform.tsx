/* eslint-disable no-alert */
/* eslint-disable no-console */
/* eslint-disable no-use-before-define */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';

import '../styles/style.css';
import { BroadcastRaw } from '../interfaces/Broadcast';

function BroadcastForm() {
  // Import useHistory for redirect functionality
  const history = useHistory();

  // Creates new joined state for all input feilds
  const [newBroadcast, setNewBroadcast] = useState<BroadcastRaw>({
    title: '',
    description: '',
    tags: '',
    owner: '',
    isReversed: false,
    youtubePlaylists: '',
  });

  // Function that saves the different inputs (by name attribute) to state
  function handleChange(evt: React.ChangeEvent<HTMLInputElement>) {
    const { value, name } = evt.target;
    setNewBroadcast({
      ...newBroadcast,
      [name]: value,
    });
  }

  // Function that saves checkbox input to state
  function handleChangeCheckbox(evt: React.ChangeEvent<HTMLInputElement>) {
    let value;
    if (evt.target.value === 'false') value = true;
    else value = false;
    setNewBroadcast({
      ...newBroadcast,
      isReversed: value,
    });
  }

  // Function to log new broadcast
  async function createBroadcast(broadcast: BroadcastRaw) {
    // Call backend API
    try {
      const response = await fetch('/api/create-broadcast', {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(broadcast), // Stringify to JSON before posting
      });
      if (response.ok) { // If response is ok (200 range)
        const broadcastObj = await response.json(); // Parse JSON response

        // Redirect user to new broadcast page
        history.push(`/b/${broadcastObj.broadcastId}`);
      } else {
        // Else if error, send to error page
        history.push('/404');
      }
    } catch (err) {
      console.log(err);
    }
  }

  return (
    <>

      <div className="broadcastFormContainer">
        <h3>Create a new broadcast</h3>

        <form
          onSubmit={(e) => {
            e.preventDefault(); /* Prevents reload of page on submit */

            // Form validation
            if (newBroadcast.title === '') { alert('Title required!'); return; }
            if (newBroadcast.owner === '') { alert('Owner field required!'); return; }
            if (newBroadcast.youtubePlaylists === '') { alert('You have not specified any youtube playlists!'); return; }
            /* Postsbroadcast to backend through createEvent function */
            createBroadcast(newBroadcast);

            /* Clears inputs */
            setNewBroadcast({
              title: '',
              description: '',
              tags: '',
              owner: '',
              youtubePlaylists: '',
              isReversed: false,
            });
          }}
        >
          <label>
            Broadcast name:
            <br />
            {/* Stores input in state onChange (everytime something is typed) */}
            <input name="title" value={newBroadcast.title} onChange={handleChange} type="text" />
          </label>
          <br />
          <br />
          <label>
            Description:
            <br />
            {/* Stores input in state onChange (everytime something is typed) */}
            <input name="description" value={newBroadcast.description} onChange={handleChange} type="text" />
          </label>
          <br />
          <br />
          <label>
            Tags:
            <br />
            {/* Stores input in state onChange (everytime something is typed) */}
            <input name="tags" value={newBroadcast.tags} onChange={handleChange} type="text" />
          </label>
          <br />
          <br />
          <label>
            Owner:
            <br />
            {/* Stores input in state onChange (everytime something is typed) */}
            <input name="owner" value={newBroadcast.owner} onChange={handleChange} type="text" />
          </label>
          <br />
          <br />
          <label>
            Playlists:
            <br />
            {/* Stores input in state onChange (everytime something is typed) */}
            <input name="youtubePlaylists" value={newBroadcast.youtubePlaylists} onChange={handleChange} type="text" />
          </label>
          <br />
          <br />
          <label>
            Want to reverse the queue order of the playlists?&nbsp;&nbsp;&nbsp;
            {/* Stores input in state onChange (everytime something is typed) */}
            <input name="isReversed" value={newBroadcast.isReversed} onChange={handleChangeCheckbox} type="checkbox" />
          </label>
          <br />
          <br />

          <button type="submit" value="Create">Create Broadcast</button>

        </form>

      </div>

    </>
  );
}

export default BroadcastForm;
