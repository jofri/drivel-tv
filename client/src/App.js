import {BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { useState } from 'react';
import './App.css';
import './styles/style.css';

import Navbar from './components/Navbar';
import Homepage from './components/Homepage';
import Broadcast from './components/Broadcast';
import Broadcastform from './components/Broadcastform';
import FourOFour from './components/404';
import DeleteForm from './components/Deleteform';



function App() {

  const [broadcast, setBroadcast] = useState({});
  const [broadcastUrl, setBroadcastUrl] = useState('/b/:broadcast');
  const [allBroadcastObjects, setAllBroadcastObjects] = useState([]);


  return (
    <Router>
      <Navbar/>
      <div className="content">
      <Switch>
        <Route exact path='/'> {/* If user visits root, redict to homepage */}
          <Homepage allBroadcasts={allBroadcastObjects} getAllBroadcasts={getAllBroadcasts}/>
        </Route>
        <Route exact path="/create-broadcast">
          <Broadcastform/>
        </Route>
        <Route exact path="/delete-broadcast">
          <DeleteForm/>
        </Route>
        <Route exact path={broadcastUrl}> {/* If user visits broadcast page, check if broadcast exists in db - else, redirect to 404*/}
          <Broadcast broadcast={broadcast} getBroadcast={getBroadcast}/>
        </Route>
        <Route exact path='/404'> {/* Specify 404 route */}
          <FourOFour/>
        </Route>
        <Route path='/'> {/* If user visits any page not specified, redirect to 404 */}
          <FourOFour/>
        </Route>
      </Switch>
      </div>
    </Router>
  )



 // Function to get all broadcasts from backend server
 async function getAllBroadcasts () {
  // Call backend API
  try {
    const response = await fetch('/api/get-all-broadcasts');
    if (response.ok) { // If response is ok (200 range)
      const allBroadcastObjects = await response.json(); // Parse JSON response
      setAllBroadcastObjects(allBroadcastObjects); // Set array of broadcast objects as state
    } else { setBroadcastUrl('/404');}; // Else if no broadcasts, send user to 404
  } catch (err) {
    console.log(err);
  }
 };

  // Function to get broadcast from backend server
  async function getBroadcast (id) {
    // Call backend API
    try {
      const response = await fetch('/api/get-broadcast', {
        method: 'POST',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({"broadcastId": id}) // Stringify to JSON before posting
      });
      if (response.ok) { // If response is ok (200 range)
        const broadcastObj = await response.json(); // Parse JSON response
        setBroadcast(broadcastObj); // Set broadcast object as state
      } else { setBroadcastUrl('/404');}; // Else if broadcast does not exist in DB, send user to 404
    } catch (err) {
      console.log(err);
    }
  };

}

export default App;
