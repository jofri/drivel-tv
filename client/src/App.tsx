/* eslint-disable no-use-before-define */
/* eslint-disable max-len */
/* eslint-disable no-console */
import React, { useState } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import './App.css';
import './styles/style.css';

import Navbar from './components/Navbar';
import Homepage from './components/Homepage';
import Broadcast from './components/Broadcast';
import Broadcastform from './components/Broadcastform';
import FourOFour from './components/404';
import DeleteForm from './components/Deleteform';
import { apiGetAllBroadcasts } from './services/apiService';
import { BroadcastInterface } from './interfaces/Broadcast';

function App() {
  const [broadcast, setBroadcast] = useState<BroadcastInterface | null>(null);
  const [broadcastUrl, setBroadcastUrl] = useState<string>('/b/:broadcast');
  const [allBroadcastObjects, setAllBroadcastObjects] = useState<BroadcastInterface[] | null>([]);

  // Refactored: Now the fetch function is in the apiServices.ts
  // now this function just: makes the API request, set the broadcast Object
  // If response is falsy, sets it to 404
  const getAllBroadcasts = async (): Promise<void> => {
    try {
      const response = await apiGetAllBroadcasts();
      if (!response) {
        setBroadcastUrl('/404');
        return;
      }
      setAllBroadcastObjects(response);
    } catch (err) {
      console.error(err);
    }
  };

  // Function to get broadcast from backend server
  async function getBroadcast(id:string) {
    // Call backend API
    try {
      const response = await fetch('/api/get-broadcast', {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ broadcastId: id }), // Stringify to JSON before posting
      });
      if (response.ok) { // If response is ok (200 range)
        const broadcastObj = await response.json(); // Parse JSON response
        setBroadcast(broadcastObj); // Set broadcast object as state
      } else { setBroadcastUrl('/404'); } // Else if broadcast does not exist in DB, send user to 404
    } catch (err) {
      console.log(err);
    }
  }
  return (
    <Router>
      <Navbar getAllBroadcasts={getAllBroadcasts} />
      <div className="content">
        <Switch>
          <Route exact path="/">
            {' '}
            {/* If user visits root, redict to homepage */}
            <Homepage allBroadcasts={allBroadcastObjects} getAllBroadcasts={getAllBroadcasts} />
          </Route>
          <Route exact path="/create-broadcast">
            <Broadcastform />
          </Route>
          <Route exact path="/delete-broadcast">
            <DeleteForm />
          </Route>
          <Route exact path={broadcastUrl}>
            {' '}
            {/* If user visits broadcast page, check if broadcast exists in db - else, redirect to 404 */}
            <Broadcast broadcast={broadcast} getBroadcast={getBroadcast} />
          </Route>
          <Route exact path="/404">
            {' '}
            {/* Specify 404 route */}
            <FourOFour />
          </Route>
          <Route path="/">
            {' '}
            {/* If user visits any page not specified, redirect to 404 */}
            <FourOFour />
          </Route>
        </Switch>
      </div>
    </Router>
  );
}

export default App;
