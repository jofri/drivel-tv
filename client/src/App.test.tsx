/* eslint-disable max-len */
/* eslint-disable no-undef */
/* eslint-disable no-use-before-define */
import {
  fireEvent, render, screen, waitFor,
} from '@testing-library/react';
import React from 'react';
import {
  BrowserRouter, MemoryRouter, Router,
} from 'react-router-dom';
import { createMemoryHistory } from 'history';
import renderer from 'react-test-renderer';
import App from './App';
import FourOFour from './components/404';
import DeleteForm from './components/Deleteform';
import Broadcastform from './components/Broadcastform';
import broadcastMock from './__mocks__/mockData';

test('renders jump to live text on the button', () => {
  render(<App />);
  const jumpToLiveButton = screen.getByText('JUMP TO LIVE');
  expect(jumpToLiveButton).toBeInTheDocument();
});
test('renders find broadcast', () => {
  render(<App />);
  const findBroadcast = screen.queryByText('Find broadcast');
  expect(findBroadcast).not.toBeInTheDocument();
});

test('full app rendering', () => {
  const history = createMemoryHistory();
  render(
    <Router history={history}>
      <App />
    </Router>,
  );
  expect(screen.getByText('Drivel.TV'));
});
test('renders correctly, snapshot', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('landing on Create Broadcast page', () => {
  const renderRoutes = (component: any, { route = '/' } = {}) => {
    window.history.pushState({}, 'CreateForm', route);
    return render(component, { wrapper: BrowserRouter });
  };
  renderRoutes(<Broadcastform />, { route: '/create-broadcast' });
  expect(screen.getByText('Create a new broadcast')).toBeInTheDocument();
});
test('landing on Delete Broadcast page', () => {
  const renderRoutes = (component: any, { route = '/' } = {}) => {
    window.history.pushState({}, 'DeleteForm', route);
    return render(component, { wrapper: BrowserRouter });
  };
  renderRoutes(<DeleteForm />, { route: '/delete-broadcast' });
  expect(screen.getByText('Delete broadcast')).toBeInTheDocument();
});
test('landing on 404 page', () => {
  const renderRoutes = (component: any, { route = '/404' } = {}) => {
    window.history.pushState({}, 'FourOFour', route);
    return render(component, { wrapper: BrowserRouter });
  };
  renderRoutes(<FourOFour />, { route: '/404' });
  expect(screen.getByText('404 - Page or Broadcast not found on server')).toBeInTheDocument();
});
test('landing on an empty page', async () => {
  const renderRoutes = (component: any, { route = '/' } = {}) => {
    window.history.pushState({}, 'Test page', route);
    return render(component, { wrapper: BrowserRouter });
  };
  renderRoutes(<App />, { route: '/something' });
  expect(screen.getByText(/404/)).toBeInTheDocument();
});
test('get all Broadcasts', () => {
  const data = broadcastMock;
  const mockFn = jest.fn().mockResolvedValue([data, data]);
  jest.mock('./services/apiService', () => ({
    __esModule: true,
    apiGetAllBroadcasts: mockFn,
  }));
  const { getAllByTestId, getByText } = render(<App />, { wrapper: MemoryRouter });
  const getLiveButton = getByText('JUMP TO LIVE');
  fireEvent.click(getLiveButton);
  waitFor(() => {
    const broadcastTiles = getAllByTestId('broadcast-tile');
    expect(broadcastTiles.length).toBe(2);
    getByText('test Title');
    expect(mockFn).toBeCalled();
  });
});
