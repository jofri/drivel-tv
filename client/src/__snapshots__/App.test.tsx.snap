// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly, snapshot 1`] = `
Array [
  <header>
    <div
      onClick={[Function]}
    >
      <p
        className="logo-text"
      >
        Drivel.TV
      </p>
    </div>
    <div
      className="search"
    >
      <form
        className="search-form"
      >
        <input
          className="search-input"
          placeholder="Find broadcast..."
          type="text"
        />
        <button
          className="search-button"
          type="button"
        >
          <img
            alt=""
            className="search-icon"
            src="search.svg"
          />
        </button>
      </form>
    </div>
    <div
      className="menu-container"
    >
      <button
        className="live-button"
        onClick={[Function]}
        type="button"
      >
        JUMP TO LIVE
      </button>
      <img
        alt=""
        className="profile-icon"
        src="guest_profile_picture.png"
      />
      <div>
        <div
          className="bm-overlay"
          onClick={[Function]}
          style={
            Object {
              "MozTransform": "translate3d(100%, 0, 0)",
              "MsTransform": "translate3d(100%, 0, 0)",
              "OTransform": "translate3d(100%, 0, 0)",
              "WebkitTransform": "translate3d(100%, 0, 0)",
              "background": "rgba(0, 0, 0, 0.3)",
              "height": "100%",
              "opacity": 0,
              "position": "fixed",
              "transform": "translate3d(100%, 0, 0)",
              "transition": "opacity 0.3s, transform 0s 0.3s",
              "width": "100%",
              "zIndex": 1000,
            }
          }
        />
        <div
          style={Object {}}
        >
          <div
            className="bm-burger-button"
            style={
              Object {
                "zIndex": 1000,
              }
            }
          >
            <button
              id="react-burger-menu-btn"
              onClick={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "background": "transparent",
                  "border": "none",
                  "cursor": "pointer",
                  "fontSize": 0,
                  "height": "100%",
                  "left": 0,
                  "margin": 0,
                  "padding": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                  "zIndex": 1,
                }
              }
            >
              Open Menu
            </button>
            <span>
              <span
                className="bm-burger-bars"
                style={
                  Object {
                    "height": "20%",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": "0%",
                  }
                }
              />
              <span
                className="bm-burger-bars"
                style={
                  Object {
                    "height": "20%",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": "40%",
                  }
                }
              />
              <span
                className="bm-burger-bars"
                style={
                  Object {
                    "height": "20%",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": "80%",
                  }
                }
              />
            </span>
          </div>
        </div>
        <div
          aria-hidden={true}
          className="bm-menu-wrap"
          id=""
          style={
            Object {
              "MozTransform": "translate3d(100%, 0, 0)",
              "MsTransform": "translate3d(100%, 0, 0)",
              "OTransform": "translate3d(100%, 0, 0)",
              "WebkitTransform": "translate3d(100%, 0, 0)",
              "height": "100%",
              "position": "fixed",
              "right": 0,
              "transform": "translate3d(100%, 0, 0)",
              "transition": "all 0.5s",
              "width": "300px",
              "zIndex": 1100,
            }
          }
        >
          <div
            className="bm-menu"
            style={
              Object {
                "boxSizing": "border-box",
                "height": "100%",
                "overflow": "auto",
              }
            }
          >
            <nav
              className="bm-item-list"
              style={
                Object {
                  "height": "100%",
                }
              }
            >
              <a
                className="bm-item menu-item"
                href="/"
                id="home"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt=""
                  src="home.svg"
                />
                Home
              </a>
              <a
                className="bm-item menu-item"
                href="/profile"
                id="profile"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt=""
                  src="profile.svg"
                />
                Profile
              </a>
              <a
                className="bm-item menu-item"
                href="/create-broadcast"
                id="create-broadcast"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt=""
                  src="broadcast.svg"
                />
                Create broadcast
              </a>
              <a
                className="bm-item menu-item"
                href="/delete-broadcast"
                id="delete-broadcast"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt=""
                  src="delete.svg"
                />
                Delete broadcast
              </a>
              <a
                className="bm-item menu-item"
                href="/log-out"
                id="log-out"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt=""
                  src="log-out.svg"
                />
                Log out
              </a>
            </nav>
          </div>
          <div
            style={Object {}}
          >
            <div
              className="bm-cross-button"
              style={
                Object {
                  "height": 24,
                  "position": "absolute",
                  "right": 8,
                  "top": 8,
                  "width": 24,
                }
              }
            >
              <button
                id="react-burger-cross-btn"
                onClick={[Function]}
                style={
                  Object {
                    "background": "transparent",
                    "border": "none",
                    "cursor": "pointer",
                    "fontSize": 0,
                    "height": "100%",
                    "left": 0,
                    "margin": 0,
                    "padding": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                    "zIndex": 1,
                  }
                }
                tabIndex={-1}
              >
                Close Menu
              </button>
              <span
                style={
                  Object {
                    "position": "absolute",
                    "right": "14px",
                    "top": "6px",
                  }
                }
              >
                <span
                  className="bm-cross"
                  style={
                    Object {
                      "height": 14,
                      "position": "absolute",
                      "transform": "rotate(45deg)",
                      "width": 3,
                    }
                  }
                />
                <span
                  className="bm-cross"
                  style={
                    Object {
                      "height": 14,
                      "position": "absolute",
                      "transform": "rotate(-45deg)",
                      "width": 3,
                    }
                  }
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>,
  <div
    className="content"
  >
     
    <div
      className="homepage"
    />
  </div>,
]
`;
